Source/WebKit/UIProcess/EnhancedSecurityTracking.cpp

```diff
@@ -35,6 +35,32 @@ namespace WebKit {
 
 using namespace WebCore;
 
+using EnhancedSecuritySitesMap = HashMap<WebCore::RegistrableDomain, EnhancedSecurityReason>;
+
+static EnhancedSecuritySitesMap& enabledSitesMap()
+{
+    static MainRunLoopNeverDestroyed<EnhancedSecuritySitesMap> staticEnabledSites;
+    return staticEnabledSites;
+}
+
+static bool didSitePreviouslyUseEnhancedSecurity(const API::Navigation& navigation)
+{
+    return enabledSitesMap().contains(RegistrableDomain { navigation.currentRequest().url() });
+}
+
+void EnhancedSecurityTracking::initializeWithWebsiteDataStore(WebsiteDataStore& websiteDataStore)
+{
+    websiteDataStore.fetchEnhancedSecurityOnlyDomains([this](HashSet<RegistrableDomain>&& domains) {
```

Comment:
I doubt safer cpp static analysis will be happy with this, assuming the lambda runs asynchronously. Capturing a raw pointer like this is hard to prove as safe. We likely want to capture a smart pointer.